packages/core/src/main/Account.ts

```diff
@@ -429,31 +437,31 @@ class Account extends EventEmitter {
       const data = await this.handleEvent(event);
       if (data) {
         switch (data.type) {
-          case GenericMessageType.ASSET:
-            this.emit(Account.INCOMING.ASSET, data);
+          case Account.INCOMING.ASSET:
+            this.emit(data.type, event);
             break;
-          case GenericMessageType.CLIENT_ACTION:
-            this.emit(Account.INCOMING.CLIENT_ACTION, data);
+          case Account.INCOMING.CLIENT_ACTION:
+            this.emit(data.type, event);
             break;
-          case GenericMessageType.CONFIRMATION:
-            this.emit(Account.INCOMING.CONFIRMATION, data);
+          case Account.INCOMING.CONFIRMATION:
+            this.emit(data.type, event);
             break;
-          case GenericMessageType.DELETED:
-            this.emit(Account.INCOMING.DELETED, data);
+          case Account.INCOMING.DELETED:
+            this.emit(data.type, event);
             break;
-          case GenericMessageType.HIDDEN:
-            this.emit(Account.INCOMING.HIDDEN, data);
+          case Account.INCOMING.HIDDEN:
+            this.emit(data.type, event);
             break;
-          case GenericMessageType.KNOCK:
-            this.emit(Account.INCOMING.PING, data);
+          case Account.INCOMING.PING:
+            this.emit(data.type, event);
             break;
-          case GenericMessageType.REACTION:
-            this.emit(Account.INCOMING.REACTION, data);
+          case Account.INCOMING.REACTION:
+            this.emit(data.type, event);
             break;
-          case GenericMessageType.TEXT:
-            this.emit(Account.INCOMING.TEXT_MESSAGE, data);
+          case Account.INCOMING.TEXT_MESSAGE:
+            this.emit(data.type, event);
```

Comment:
Since they all have the same emit data, you can just write them as 
```ts
case Account.INCOMING.ASSET:
case Account.INCOMING.CLIENT_ACTION:
case Account.INCOMING.CONFIRMATION:
...
  this.emit(data.type, event);
  break;
```

Reply:
Even better, I can use the `default` to do this. :)